packages/js/components/src/link/index.tsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { partial } from 'lodash';
+import { createElement } from '@wordpress/element';
+import { getHistory } from '@woocommerce/navigation';
+import React from 'react';
+
+interface LinkProps {
+	/** Type of link. For wp-admin and wc-admin, the correct prefix is appended. */
+	type?: 'wc-admin' | 'wp-admin' | 'external';
+	/** The resource to link to. */
+	href: string;
+}
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type LinkOnClickHandlerFunction = ( ...props: any ) => any; // we don't want to restrict this function at all
+type LinkOnclickHandler = (
+	onClick: LinkOnClickHandlerFunction | undefined,
+	event: React.MouseEvent< Element > | undefined
+) => void;
+
+/**
+ * Use `Link` to create a link to another resource. It accepts a type to automatically
+ * create wp-admin links, wc-admin links, and external links.
+ */
+export const Link = ( {
+	href,
+	children,
+	type = 'wc-admin',
+	...props
+}: React.AnchorHTMLAttributes< HTMLAnchorElement > &
+	LinkProps ): React.ReactElement => {
```

Comment:
@rjchow What do you think if we use `React.FC`  instead?

Reply:
We'd get type errors on the `<a {...passProps} />` when consumers try to put on props to passthrough that are valid for a but not specified on our LinkProps 😀

Reply:
Oh, I see.